### ISSUE TITLE
engine.py breaks at lowpass_filtering and thus skipping DynamicFitter

### ISSUE THREAD

**Issue by TheOne-1** [keenon/AddBiomechanics#234]
For the AddB dataset, we removed joint angle constraints to avoid incontinuous joint angles (e.g., jumping from -pi to pi). However, this lowpass filtering script [code](https://github.com/keenon/AddBiomechanics/blob/6c9f0340c8e6a503d15d778610b70b93720d073c/server/engine/src/trial.py#L412) requires joint angle ranges to be within joint limits. As a result, DynamicFitting was not performed.

Search "Skipping dynamics fit of segment starting" in the attached log for details.

**Comment by nickbianco** [comment]
Thanks for raising this @TheOne-1. Could you clarify one thing: so instead of jumping from -pi to pi, the joint angle can go from -pi to -2pi and beyond?

**Comment by TheOne-1** [comment]
Yes Keenon suggested allowing +-2pi and beyond angles to avoid discontinuity before filtering. Accordingly I modified osim limits to -10pi ~ 10pi.

**Comment by nickbianco** [comment]
And that fixed your problem? Or do we need to add a fix here?

**Comment by TheOne-1** [comment]
After digging more it looks like the issue has nothing to do with joint limits in osim model.

In theory [this function](https://github.com/keenon/AddBiomechanics/blob/6c9f0340c8e6a503d15d778610b70b93720d073c/server/engine/src/trial.py#L406C35-L406C58) should take care of joint angle wrapping, but it failed. Maybe I should raise it in the Nimble repo.

**Comment by nickbianco** [comment]
If you're able to reproduce the bug in an issue on nimblephysics, that would be great! (Feel free to close this issue if it is no longer needed).
